lightning/backend/local.go

```diff
@@ -406,11 +484,13 @@ func (local *local) getGrpcConnLocked(ctx context.Context, storeID uint64) (*grp
 
 // Close the importer connection.
 func (local *local) Close() {
-	local.engines.Range(func(k, v interface{}) bool {
-		v.(*LocalFile).Close()
-		return true
-	})
+	allEngines := local.lockAllEnginesUnless(importMutexStateClose, 0)
```

Comment:
some functions like `AppendRows` didn't check context cancel, I'm afraid that some parent goroutine exits because of context cancel (like line 2429 in restore.go) while the child goroutine is still running, which may cause a "false" 0 in time slice between unlocking and locking.